File: src/main/java/net/onelitefeather/titan/function/DeathFunction.java
package net.onelitefeather.titan.function;

import net.kyori.adventure.text.Component;
import net.minestom.server.event.Event;
import net.minestom.server.event.EventNode;
import net.minestom.server.event.player.PlayerDeathEvent;

public final class DeathFunction {


    public DeathFunction(EventNode<Event> node) {
        node.addListener(PlayerDeathEvent.class, this::onDeath);
    }

    private void onDeath(final PlayerDeathEvent event) {
        event.setDeathText(Component.empty());
        event.getPlayer().respawn();
    }

    public static DeathFunction instance(EventNode<Event> node) {
        return new DeathFunction(node);
    }
}

File: src/main/kotlin/net/onelitefeather/titan/function/DeathFunction.kt
package net.onelitefeather.titan.function

import com.google.inject.Inject
import com.google.inject.name.Named
import net.kyori.adventure.text.Component
import net.minestom.server.event.Event
import net.minestom.server.event.EventNode
import net.minestom.server.event.player.PlayerDeathEvent

class DeathFunction: TitanFunction {

    @Named("titanNode")
    @Inject
    lateinit var eventNode: EventNode<Event>

    private fun deathListener(event: PlayerDeathEvent) {
        event.deathText = Component.empty()
        event.player.respawn()
    }

    override fun initialize() {
        eventNode.addListener(
            PlayerDeathEvent::class.java,
            this::deathListener
        )
    }

    override fun terminate() {
    }
}